src/GoogleAuthenticator/Helper.php

```diff
@@ -30,13 +32,36 @@ class Helper
     protected $authenticator;
 
     /**
-     * @param string                  $server
-     * @param BaseGoogleAuthenticator $authenticator
+     * @var string[]
      */
-    public function __construct($server, BaseGoogleAuthenticator $authenticator)
-    {
+    private $forcedForRoles;
+
+    /**
+     * @var string[]
+     */
+    private $ipWhiteList;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @param array $forcedForRole Roles that require 2FA authorization
+     * @param array $ipWhiteList   IPs that will bypass 2FA authorization
+     */
+    public function __construct(
+      $server,
+      array $forcedForRoles,
+      array $ipWhiteList,
```

Comment:
Why do you want to put those in first position? Optional and last is best IMO

Reply:
It's mandatory to make that optional and last if you want to be BC

Reply:
@greg0ire why it's need to be optional? It has default values in configuration.

Reply:
It needs to be optional to allow the constructor to be BC, you don't know if someone builds this class manually or not.